### PR TITLE
Part 4: Publish Workflow

### PR DESCRIPTION
**Pull request by RowbDowg**
This PR adds the ability to publish when a commit is tagged with a version.
The version number for the library is now also determined based off the latest tag.

### CODE REVIEW

**Review by matthewelwell — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  package:
+    name: Publish Gradle Dependency
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Preconfigure gradle
+        uses: ./.github/actions/prepare-gradle
+      - name: Unit tests
```
Comment (matthewelwell) [inline .github/workflows/publish.yml:17]: I think the name here is incorrect?

**Review by matthewelwell — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (RowbDowg) [reply .github/workflows/publish.yml]: Indeed it was. Fixed now

**Review by matthewelwell — APPROVED** [review]

**Comment by matthewelwell** [comment]
@RowbDowg this looks good - any objects to merging it?

**Comment by RowbDowg** [comment]
@matthewelwell Nope. Go ahead and merge.

**Pull request merged**